src/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -284,5 +284,9 @@ func (dsInfo *builderInfo) unprivilegedSecurityContext() *corev1.SecurityContext
 		securityContext.RunAsUser = &unprivilegedUser
 		securityContext.RunAsGroup = &unprivilegedGroup
 	}
+	runAsPrivileged := true
+	if dsInfo.instance.OneAgentRunsPrivileged() {
+		securityContext.Privileged = &runAsPrivileged
```

Comment:
not sure how this works with running as unprivileged user/group. maybe only providing a privileged securitycontext in this case and skipping the rest would be best